internal/utils/main.go

```diff
@@ -240,10 +242,115 @@ func MustExportTypeFlags(flags *pflag.FlagSet, logger *log.Entry) (exportAccount
 	return
 }
 
-// CreateBackend creates a history archive backend
-func CreateBackend() (*ledgerbackend.HistoryArchiveBackend, error) {
+type historyArchiveBackend struct {
+	client  historyarchive.ArchiveInterface
+	ledgers map[uint32]*historyarchive.Ledger
+}
+
+func (h historyArchiveBackend) GetLatestLedgerSequence() (sequence uint32, err error) {
+	root, err := h.client.GetRootHAS()
+	if err != nil {
+		return 0, err
+	}
+	return root.CurrentLedger, nil
+}
+
+func (h historyArchiveBackend) GetLedger(sequence uint32) (bool, xdr.LedgerCloseMeta, error) {
+	ledger, ok := h.ledgers[sequence]
+	if !ok {
+		return false, xdr.LedgerCloseMeta{}, fmt.Errorf("ledger %d is missing from map", sequence)
+	}
+
+	lcm := xdr.LedgerCloseMeta{
+		V: 0,
+		V0: &xdr.LedgerCloseMetaV0{
+			LedgerHeader: ledger.Header,
+			TxSet:        ledger.Transaction.TxSet,
+		},
+	}
+	lcm.V0.TxProcessing = make([]xdr.TransactionResultMeta, len(ledger.TransactionResult.TxResultSet.Results))
+	for i, result := range ledger.TransactionResult.TxResultSet.Results {
+		lcm.V0.TxProcessing[i].Result = result
+	}
+
+	return true, lcm, nil
+}
+
+func (h historyArchiveBackend) PrepareRange(ledgerbackend.Range) error {
+	return nil
+}
+
+func (h historyArchiveBackend) IsPrepared(ledgerbackend.Range) (bool, error) {
+	return true, nil
+}
+
+func (h historyArchiveBackend) Close() error {
+	return nil
+}
+
+func ValidateLedgerRange(start, end, latestNum uint32) error {
+	if start == 0 {
+		return fmt.Errorf("Start sequence number equal to 0. There is no ledger 0 (genesis ledger is ledger 1)")
+	}
+
+	if end == 0 {
+		return fmt.Errorf("End sequence number equal to 0. There is no ledger 0 (genesis ledger is ledger 1)")
+	}
+
+	if end < start {
+		return fmt.Errorf("End sequence number is less than start (%d < %d)", end, start)
+	}
+
+	if latestNum < start {
+		return fmt.Errorf("Latest sequence number is less than start sequence number (%d < %d)", latestNum, start)
+	}
+
+	if latestNum < end {
+		return fmt.Errorf("Latest sequence number is less than end sequence number (%d < %d)", latestNum, end)
+	}
+
+	return nil
+}
+
+func CreateBackend(start, end uint32) (ledgerbackend.LedgerBackend, error) {
+	client, err := CreateHistoryArchiveClient()
+	if err != nil {
+		return nil, err
+	}
+
+	root, err := client.GetRootHAS()
+	if err != nil {
+		return nil, err
+	}
+	if err = ValidateLedgerRange(start, end, root.CurrentLedger); err != nil {
+		return nil, err
+	}
+
+	ledgers, err := client.GetLedgers(start, end)
+	if err != nil {
+		return nil, err
+	}
+	return historyArchiveBackend{client: client, ledgers: ledgers}, nil
+}
+
+func CreateHistoryArchiveClient() (historyarchive.ArchiveInterface, error) {
 	archiveStellarURL := "http://history.stellar.org/prd/core-live/core_live_001"
```

Comment:
Not in a scope of this PR but this should be `https` because tx data is not signed.
```suggestion
	archiveStellarURL := "https://history.stellar.org/prd/core-live/core_live_001"
```

Reply:
Likewise not in scope, but using the new `historyarchive.NewArchivePool` (stellar/go#3402) with multiple URLs here could be useful here for resilience.